identity_derive/src/utils.rs

```diff
@@ -0,0 +1,73 @@
+use proc_macro2::{Delimiter, TokenTree};
+
+use syn::{DeriveInput, Field, Meta, MetaList, NestedMeta, Path, PathSegment};
+
+use quote::format_ident;
+
+const PARENS: Delimiter = Delimiter::Parenthesis;
+
+/// checks to see if a field's type is `Option`.  This logic is necessary to find cases where fields contain nested
+/// Options and avoid a `Some(None)` case.
+pub fn extract_option_segment(path: &Path) -> Option<&PathSegment> {
```

Comment:
`path.segments.last().filter(|segment| segment == "Option")`

I think this does the same and avoids the two allocations?

Reply:
Issue is that the path segments need to be converted into a `String` before they can be compared to a `&str` since theres no `PartialEq` for `PathSegment` to `&str` or `String`.